modules/database/src/ioc/db/dbAccess.c

```diff
@@ -1369,8 +1370,22 @@ long dbPut(DBADDR *paddr, short dbrType,
         if (nRequest < 1) {
             recGblSetSevr(precord, LINK_ALARM, INVALID_ALARM);
         } else {
-            status = dbFastPutConvertRoutine[dbrType][field_type](pbuffer,
-                paddr->pfield, paddr);
+            if (propertyUpdate && paddr->field_size <= sizeof(epicsAny)) {
+                epicsAny propBuffer;
```

Comment:
Oops, I did not check what `epicsString` actually is and assumed 40 chars (confusing it with `epicsOldString`). It only worked "coincidentally" when I tested it with .EGU because I used a 64 bit machine.

Reply:
Small problem with `MAX_ENUM_STRING_SIZE`: It is part of the CA definitions (db_access.h) which notoriously disagree with the rest of EPICS (e.g. compare `MAX_UNITS_SIZE` 8 with EGU size of 16 for records in base). We also cannot be sure that some record outside base would never use longer string property fields. Thus, I prefer to use `MAX_STRING_SIZE`(i.e. 40 bytes) here. I don't think the additional 14 bytes of stack use will hurt us.

Reply:
I just noticed that mbbi/o records handle `DBE_PROPERTY` in their `special()` function and need to be fixed as well (unlike bi/o records).

Reply:
Ack, sorry my bad; `MAX_STRING_SIZE` is the correct value for the "new" dbAccess routines, as shown here in dbAccessDefs.h:
```C
#define DBRenumStrs \
        epicsUInt32     no_str;         /* number of strings*/\
        epicsInt32      padenumStrs;    /*padding to force 8 byte align*/\
        char            strs[DB_MAX_CHOICES][MAX_STRING_SIZE];  /* string values    */
```